tendermint/src/public_key.rs

```diff
@@ -3,7 +3,9 @@
 use crate::error::{Error, Kind};
 use anomaly::fail;
 use serde::{de::Error as _, Deserialize, Deserializer, Serialize, Serializer};
-use signatory::{ecdsa::curve::secp256k1, ed25519};
+#[cfg(feature = "secp256k1")]
```

Comment:
If we moved everything secp-related into a separate file/module, we can likely just annotate it on the `mod` level instead? Not blocking this PR but probably worth separating this out?